images/linux/scripts/base/snap.sh

```diff
@@ -5,4 +5,10 @@
 # when they are rolling a new major update out.
 # Hold is calculated as today's date + 60 days
 
+# snapd is started automatically, but during image generation
+# a unix socket may die, restart snapd.service (and therefore snapd.socket)
+# to make sure the socket is alive.
+
+systemctl restart snapd.socket
```

Comment:
Should we check if the socket exists before restart?

Reply:
@al-cheb this socket exists in anyway i.e. it just a different type of systemd unit provided by the snapd package. In short snapd.socket is not the unix socket I am referring to above it is just a service that polls snapd daemon from time to time